### PR TITLE
Change zen service to zen server

### PR DESCRIPTION
**Pull request by safoinme**
## Describe changes

- Change the name "Zen Service/ZenService" to "Zen Server/ZenServer". _**ENG-875**_
- Remove import and handle registration of the ZenServer "ZenService" more gracefully. _**ENG-863**_

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [ ] If my change requires a change to docs, I have updated the documentation accordingly.
- [X] I have added tests to cover my changes.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Other (add details above)

### CODE REVIEW

File: src/zenml/cli/config.py [hunk 1]
```diff
@@ -123,7 +123,7 @@ def profile() -> None:
 @click.option(
     "--user",
     "user_name",
-    help="The username that is used to authenticate with the ZenService. This "
+    help="The username that is used to authenticate with the ZenServe. This "
```
Comment (stefannica) [inline src/zenml/cli/config.py:126]:
```suggestion
    help="The username that is used to authenticate with the ZenServer. This "
```

File: src/zenml/cli/server.py [hunk 2]
```diff
@@ -30,112 +30,112 @@
...
-def service() -> None:
+def server() -> None:
     """ZenML server."""
 
 
-@service.command("explain", help="Explain the service")
-def explain_service() -> None:
-    """Explain the concept of the Zen Service."""
-    component_module = import_module("zenml.zen_service")
+@server.command("explain", help="Explain the the concept of the Zen Server.")
```
Comment (stefannica) [inline src/zenml/cli/server.py:47]:
```suggestion
@server.command("explain", help="Explain the Zen Server concept.")
```

File: src/zenml/cli/server.py [hunk 3]
```diff
@@ -30,112 +30,112 @@
...
+def explain_server() -> None:
+    """Explain the concept of the Zen Server."""
```
Comment (stefannica) [inline src/zenml/cli/server.py:49]:
```suggestion
    """Explain the Zen Server concept."""
```

File: src/zenml/zen_server/zen_server.py [hunk 4]
```diff
@@ -144,17 +144,17 @@ def run(self) -> None:
...
             )
         except KeyboardInterrupt:
-            logger.info("Zen service stopped. Resuming normal execution.")
+            logger.info("Zen Server stopped. Resuming normal execution.")
 
     @property
-    def zen_service_uri(self) -> Optional[str]:
-        """Get the URI where the service is running.
+    def zen_server_uri(self) -> Optional[str]:
+        """Get the URI where the service responisble for Zen Server is running.
```
Comment (stefannica) [inline src/zenml/zen_server/zen_server.py:157]:
```suggestion
        """Get the URI where the service responsible for Zen Server is running.
```

**Review by stefannica — CHANGES_REQUESTED** [review]
Found a few typos, but looks good otherwise.

File: src/zenml/cli/server.py [hunk 5]
```diff
@@ -30,112 +30,112 @@
...
 )
 
 
 @cli.group(
     cls=TagGroup,
     tag=CliCategories.MANAGEMENT_TOOLS,
 )
-def service() -> None:
+def server() -> None:
     """ZenML server."""
```
Comment (strickvl) [inline src/zenml/cli/server.py:44]: Should this be ZenServer?

Comment (strickvl) [reply src/zenml/cli/server.py:44]: I guess this question applies to any other place where we talk about 'ZenML server'.

File: src/zenml/cli/server.py [hunk 6]
```diff
@@ -30,112 +30,112 @@
...
         console.print(md)
 
 
-@service.command("up", help="Start a daemon service running the zenml service")
+@server.command("up", help="Start a daemon service running the ZenServer.")
 @click.option("--port", type=int, default=8000, show_default=True)
 @click.option("--profile", type=str, default=None)
 def up_server(port: int, profile: Optional[str]) -> None:
-    """Provisions resources for the zen service."""
+    """Provisions resources for the zen server."""
```
Comment (strickvl) [inline src/zenml/cli/server.py:61]:
```suggestion
    """Provisions resources for the ZenServer."""
```

File: src/zenml/zen_server/zen_server.py [hunk 7]
```diff
@@ -26,44 +26,44 @@
...
 
 
-class ZenServiceEndpointConfig(LocalDaemonServiceEndpointConfig):
-    """Zen Service endpoint configuration.
+class ZenServerEndpointConfig(LocalDaemonServiceEndpointConfig):
+    """ZenServer endpoint configuration.
 
     Attributes:
-        zen_service_uri_path: URI path for the zenml service
+        zen_server_uri_path: URI path for the zenml service
```
Comment (strickvl) [inline src/zenml/zen_server/zen_server.py:37]:
```suggestion
        zen_server_uri_path: URI path for the ZenServer
```

?

File: src/zenml/zen_server/zen_server.py [hunk 8]
```diff
@@ -26,44 +26,44 @@
...
 
 
-class ZenServiceConfig(LocalDaemonServiceConfig):
-    """Zen Service deployment configuration.
+class ZenServerConfig(LocalDaemonServiceConfig):
+    """ZenServer deployment configuration.
 
     Attributes:
-        port: Port at which the service is running
+        port: Port at which the service responisble for ZenServer is running
```
Comment (strickvl) [inline src/zenml/zen_server/zen_server.py:66]:
```suggestion
        port: Port at which the service responsible for the ZenServer is running
```

File: src/zenml/zen_server/zen_server.py [hunk 9]
```diff
@@ -144,17 +144,17 @@ def run(self) -> None:
...
             )
         except KeyboardInterrupt:
-            logger.info("Zen service stopped. Resuming normal execution.")
+            logger.info("ZenServer stopped. Resuming normal execution.")
 
     @property
-    def zen_service_uri(self) -> Optional[str]:
-        """Get the URI where the service is running.
+    def zen_server_uri(self) -> Optional[str]:
+        """Get the URI where the service responsible for ZenServer is running.
```
Comment (strickvl) [inline src/zenml/zen_server/zen_server.py:157]:
```suggestion
        """Get the URI where the service responsible for the ZenServer is running.
```

File: tests/unit/zen_server/test_zen_server.py [hunk 10]
```diff
@@ -20,14 +20,14 @@
...
+from zenml.zen_server.zen_server import ZenServer, ZenServerConfig
 from zenml.zen_stores import LocalZenStore
 from zenml.zen_stores.base_zen_store import DEFAULT_USERNAME
 
 
 @pytest.fixture
-def running_zen_service(tmp_path_factory: pytest.TempPathFactory) -> ZenService:
-    """Spin up a zen service to do tests on."""
+def running_zen_server(tmp_path_factory: pytest.TempPathFactory) -> ZenServer:
+    """Spin up a zen server to do tests on."""
```
Comment (strickvl) [inline tests/unit/zen_server/test_zen_server.py:30]:
```suggestion
    """Spin up a ZenServer for tests."""
```

**Review by strickvl — CHANGES_REQUESTED** [review]
Some small nits in here. I think you should make sure that changes happening in the various other docs branches also reflect this naming change (or maybe just message the people working on that to make sure they're aware). Otherwise looks good to me!

**Review by strickvl — APPROVED** [review]
🙌

**Pull request merged**